msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/SignUpStates.kt

```diff
@@ -609,89 +639,103 @@ class SignUpAttributesRequiredState internal constructor(
+                                continuationToken = result.continuationToken,
+                                correlationId = result.correlationId,
+                                username = username,
+                                config = config
+                            ),
+                            requiredAttributes = result.requiredAttributes.toListOfRequiredUserAttribute()
+                        )
+                    }
+
+                    is SignUpCommandResult.Complete -> {
+                        SignUpResult.Complete(
+                            nextState = SignInContinuationState(
+                                continuationToken = result.continuationToken,
+                                correlationId = result.correlationId,
+                                username = username,
+                                config = config
+                            )
+                        )
+                    }
 
-            val rawCommandResult = CommandDispatcher.submitSilentReturningFuture(command).get()
+                    is SignUpCommandResult.InvalidAttributes -> {
+                        SignUpSubmitAttributesError(
+                            errorType = SignUpErrorTypes.INVALID_ATTRIBUTES,
+                            error = result.error,
+                            errorMessage = result.errorDescription,
+                            correlationId = result.correlationId
+                        )
+                    }
 
-            return@withContext when (val result = rawCommandResult.checkAndWrapCommandResultType<SignUpSubmitUserAttributesCommandResult>()) {
-                is SignUpCommandResult.AttributesRequired -> {
-                    SignUpResult.AttributesRequired(
-                        nextState = SignUpAttributesRequiredState(
-                            continuationToken = result.continuationToken,
-                            correlationId = result.correlationId,
-                            username = username,
-                            config = config
-                        ),
-                        requiredAttributes = result.requiredAttributes.toListOfRequiredUserAttribute()
-                    )
-                }
-                is SignUpCommandResult.Complete -> {
-                    SignUpResult.Complete(
-                        nextState = SignInContinuationState(
-                            continuationToken = result.continuationToken,
+                    is INativeAuthCommandResult.Redirect -> {
+                        SignUpSubmitAttributesError(
+                            errorType = ErrorTypes.BROWSER_REQUIRED,
+                            error = result.error,
+                            errorMessage = result.errorDescription,
+                            correlationId = result.correlationId
+                        )
+                    }
+                    // This should be caught earlier in the flow, so throwing UnexpectedError
```

Comment:
nit: let's update this comment: `// This should be caught earlier in the flow, so returning generic error`